### ISSUE TITLE
MetadataService doesn't update Package when updating / adding EntityTypes to package

### ISSUE THREAD

**Issue by fdlk** [molgenis/molgenis#7264]
The "rule" for one to many relations, including the Package <-> EntityType relation is that when updating the side with the xref (i.e. EntityType), you also need to do an update of the side with the one-to-many (i.e. Package).

### Code inspection
MetaDataServiceImpl.upsertEntityTypesSkipMappedByAttributes only does an update or add of the EntityType, and doesn't update the package(s) affected. This means that the PackageRepositoryDecorator cannot enforce security constraints on the packages.

**Comment by tommydeboer** [comment]
@fdlk `upsertEntityTypesSkipMappedByAttributes` is step 1, followed by `addMappedByAttributes`. The second step updates the Entity Type with the package attribute, wouldn't that also update the Package entity?

**Comment by fdlk** [comment]
@tommydeboer Not automatically, only the repository that you do the update call on gets updated.

**Comment by tommydeboer** [comment]
I can't reproduce it in any way. There are two places this method is used: during bootstrapping and in the metadata manager. 

For the bootstrapping, I tried putting `FileIngestJobExecutionMetadata` in the system package. That works, and the Package entities get updated as you would expect.

In the metadata manager I tried selecting different packages for an entity type, and in each case the Package entities were updated correctly.
